src/_nebari/utils.py

```diff
@@ -314,3 +314,39 @@ def construct_azure_resource_group_name(
     if base_resource_group_name:
         return f"{base_resource_group_name}{suffix}"
     return f"{project_name}-{namespace}{suffix}"
+
+
+def get_k8s_version_prefix(k8s_version: str) -> str:
+    """Return the major.minor version of the k8s version string."""
+
+    k8s_version = str(k8s_version)
+    # Split the input string by the first decimal point
+    parts = k8s_version.split(".", 1)
+
+    if len(parts) == 2:
+        # Extract the part before the second decimal point
+        before_second_decimal = parts[0] + "." + parts[1].split(".")[0]
+        try:
+            # Convert the extracted part to a float
+            result = float(before_second_decimal)
+            return result
+        except ValueError:
+            # Handle the case where the conversion to float fails
+            return None
+    else:
+        # Handle the case where there is no second decimal point
+        return None
```

Comment:
you can use `packaging` for this:

```python
>>> from packaging import version
>>> version.parse('2.3.4')
<Version('2.3.4')>


>>> version.parse('2.3.4') > version.parse('2.3.1')
True
```

Reply:
The DO versions aren't just numeric (example -  `1.18.19-do.0`) and I get back `packaging.version.InvalidVersion` when trying to use this library for them

Reply:
That's a slug, not the actual k8s versions, you can get explicit kubernetes versions too, e.g.:

```bash
$ doctl kubernetes options versions

Slug            Kubernetes Version    Supported Features
1.28.2-do.0     1.28.2                cluster-autoscaler, docr-integration, ha-control-plane, token-authentication
1.27.6-do.0     1.27.6                cluster-autoscaler, docr-integration, ha-control-plane, token-authentication
1.26.9-do.0     1.26.9                cluster-autoscaler, docr-integration, ha-control-plane, token-authentication
1.25.14-do.0    1.25.14               cluster-autoscaler, docr-integration, ha-control-plane, token-authentication
```

Alternatively you can split by `-` and use the first part.